File: book/03.ts
// 동명이인 찾기

let names: string[] = ["Tom", "Jerry", "Mike", "Tom", "Jerry", "Mike"];

function findSameName(names: string[]) {
  let same = new Set();
  for (let i = 0; i < names.length - 1; i++) {
    for (let j = i + 1; j < names.length; j++) {
      if (names[i] === names[j]) {
        same.add(names[i]);
      }
    }
  }
  return same;
}

console.log(findSameName(names));

File: book/test/03.py
# 동명이인 찾기

arr = ["Tom", "Jerry", "Mike"]


def find_same_name(arr):
    same_list = []
    n = len(arr)
    for i in range(n-1):
        for j in range(i+1, n):
            if arr[i] == arr[j] and arr[i] not in same_list:
                same_list.append(arr[i])
    return same_list


def print_each(arr):
    for i in arr:
        print(i)


# print_each(find_same_name(arr))

def find_same_set(arr):
    same_set = set()
    n = len(arr)
    for i in range(n-1):
        for j in range(i+1, n):
            if arr[i] == arr[j]:
                same_set.add(arr[i])
    return same_set


# print_each(find_same_set(arr))
print(find_same_set(arr))
